main.py

```diff
@@ -0,0 +1,77 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import schemas
+from crud import (
+    get_all_authors,
+    create_author,
+    get_author_detail,
+    get_all_books,
+    create_book,
+)
+from database import SessionLocal
+
+
+app = FastAPI()
+
+API_PREFIX = "/api/v1/"
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
```

Comment:
This function `get_db` is a dependency that provides a database session for each request and closes it after the request is finished. It's a good practice to handle the database session this way in FastAPI. However, the function could be moved to a separate file (for example, `dependencies.py`) to keep the main file clean and focused on routing.